test/function_compilation.jl

```diff
@@ -4,16 +4,18 @@ using Test
 using DataFramesMeta
 
 @testset "function_compilation" begin
+    @eval begin
```

Comment:
what is the benefit of evaluating it on module level?

Reply:
The stuff inside the `for` loop needs to be `@eval`ed to prevent Julia from caching the anonymous functions inside the `for` loop. 

The `for` loop needs to happen to test that we compare the compiled times for the macros which should be `:x => f => :y` where `f` is the same and thus added to the method table. 

This logic is complicated and realistically I think we should only be checking the timings on the 2nd run, so we should add `if i == 2` somewhere. But it's working now and I am a bit scared to mess with it in this PR.

Reply:
OK